src/Interpreters/InterserverCredentials.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
```

Comment:
>Should I move the code somewhere other than Interpreters?

Why not put it near the `InterserverIOHTTPHandler.cpp`?

Reply:
It's used from shared (replicated* and also interserver handler). Closer to the handler makes more sense for handler but not client. what do you think about storing client + server closer to usage? (Moving client creds back to shares)

Reply:
Not sure that I got this, but now I noticed that it is used in the Interpreters too, so looks ok to me as-is